### PR TITLE
update zoekt on sourcegraph/sourcegraph

### LINKED ISSUE #31181: devx: Continous delivery of Zoekt
Zoekt currently requires manual deployment. This has led to large groups of commits to accumulate which caused hard to debug production incidents. We'd like to change Zoekt's CI to automatically run [`./dev/zoekt/update`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/dev/zoekt/update?subtree=true) eagerly after a successful master build.

### PR DESCRIPTION
**Pull request by davejrt**
updates the go.mod/go.sum on sourcegraph/sourcegraph when a new commit has been merged to `main`. 

Fixes https://github.com/sourcegraph/sourcegraph/issues/31181

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+jobs: 
+  sync-zoekt:
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v2
+        with: 
+          repository: 'sourcegraph/sourcegraph' 
+          ref: 'main'
+      - uses: actions/setup-go@v2
+        name: 'Update zoekt'
```
Comment (bobheadxi) [inline .github/workflows/pr.yml]:
```suggestion
```

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - run: ./dev/zoekt/update
+      - uses: peter-evans/create-pull-request@v3
+        name: 'Create PR'
+        id: pr
+        with: 
+          token: ${{ secrets.GH_TOKEN }}
+          title: Update Zoekt
+          commit-message: 'update to latest version of zoekt'
+          body: |
+            This PR is autogenerated to update zoekt.
```
Comment (bobheadxi) [inline .github/workflows/pr.yml]:
```suggestion
          title: 'zoekt: update to sourcegraph/zoekt@${{ github.sha }}'
          commit-message: 'zoekt: update to sourcegraph/zoekt@${{ github.sha }}'
          body: |
            This PR is autogenerated to update to sourcegraph/zoekt@${{ github.sha }}
```

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+            ## Test Plan
+
+            Testing is done on the zoekt repo. All checks on this PR are sufficient.
+
+          branch: 'update-zoekt/zoekt'
```
Comment (bobheadxi) [inline .github/workflows/pr.yml]:
```suggestion
          branch: 'zoekt/update'
```

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          delete-branch: 'true'
+          reviewers: 'sourcegraph/search-core'
+          base: 'main'
+          labels: 'automerge'
```
Comment (bobheadxi) [inline .github/workflows/pr.yml]:
```suggestion
```

Looks like we have enable "real" automerge below, so we shouldn't need a label on top

File: .github/workflows/pr.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+name: sync zoekt
+on:
+  push:
```
Comment (bobheadxi) [inline .github/workflows/pr.yml:3]: Can we add a manual dispatch and create a test PR from this? https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/

Comment (davejrt) [reply .github/workflows/pr.yml:3]: would the intention be to then close the test PR as part of the same work flow?

Comment (bobheadxi) [reply .github/workflows/pr.yml:3]: No need - you can just create it, and if nobody reviews it, it should just stay open right? That would be an effective test run perhaps

**Comment by bobheadxi** [comment]
@davejrt looks like https://github.com/sourcegraph/sourcegraph/pull/32286 is the test PR right? looks great! 😁

**Review by bobheadxi — APPROVED** [review]
very nice!

**Comment by keegancsmith** [comment]
@davejrt this should only run once `go test` and the docker image building steps have completed. Otherwise the PR we create in sourcegraph will fail when trying to build the zoekt docker image. This has the potential bad side effect where we don't test the image in the PR workflow, but only once merged -> broken main.

**Pull request merged**